### PR TITLE
fix sample-controller klog not initflag

### LINKED ISSUE #36: How to get the log message ?
I am getting error for the log messages:

```
# ./sample-controller -kubeconfig=$HOME/.kube/config -logtostderr=true -v=2
flag provided but not defined: -logtostderr
Usage of ./sample-controller:
  -kubeconfig string
        Path to a kubeconfig. Only required if out-of-cluster.
  -master string
        The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.
```

How to check it ?

### PR DESCRIPTION
**Pull request by liucimin**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

The pr fix the sample controller no log output bug.
It because the main function not init the klog's Flags.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/sample-controller/issues/36

**Does this PR introduce a user-facing change?**:

NONE

**Release note**:

```
NONE

```

### CODE REVIEW

File: staging/src/k8s.io/sample-controller/main.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ var (
 )
 
 func main() {
+        //Init the klog flags
```
Comment (nikhita) [inline staging/src/k8s.io/sample-controller/main.go]: I think this comment can be removed (doesn't add much) but lgtm otherwise.

Comment (liucimin) [reply staging/src/k8s.io/sample-controller/main.go]: ok, I will  remove this comment.
Thx for your reply.

**Comment by liucimin** [comment]
@nikhita 
Would do you like to review it again?
Thx.

**Comment by nikhita** [comment]
Just realized we initialize some flags here:

https://github.com/kubernetes/kubernetes/blob/c5242f0368eefbdab83ad243fc14bc4001f0c5cc/staging/src/k8s.io/sample-controller/main.go#L78-L81

So InitFlags shouldn't have nil arguments.

/cc @dims 
to sanity check klog stuff

**Comment by liucimin** [comment]
@nikhita 
If we pass the nil in the klog.InitFlags()
It will use the default flagset " flag.CommandLine".
And it is the same flagset which we use in 

func init() { 
        flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.") 
        flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.") 
 }

So, i think pass nil will help us to fix this issue.
thx

**Comment by roycaihw** [comment]
`klog.InitFlags(nil)` does use the same flagset so it LGTM

@dims we have a couple of main packages missing `klog.InitFlags` besides this one: 
- [sample-apiserver](https://github.com/kubernetes/kubernetes/blob/91bc8095e553855ab54828d5aa2426a8f660f92a/staging/src/k8s.io/sample-apiserver/main.go#L30)
- [sample webhook-server](https://github.com/kubernetes/kubernetes/pull/74562/files#diff-c5322bc00350ce0e4847295d06fc1157R139) I recently fixed

it would be great if you can sanity check if it's intentional

**Comment by liucimin** [comment]
friendly ping @nikhita

**Pull request closed without merging**